sklearnex/tests/test_common.py

```diff
@@ -52,3 +57,38 @@ def test_target_offload_ban():
 
     output = "\n".join(output)
     assert output == "", f"sklearn versioning is occuring in: \n{output}"
+
+
+def _sklearnex_walk(func):
+    """this replaces checks on pkgutils to look through sklearnex
+    folders specifically"""
+
+    def wrap(*args, **kwargs):
+        if "prefix" in kwargs and kwargs["prefix"] == "sklearn.":
+            kwargs["prefix"] = "sklearnex."
+        if "path" in kwargs:
+            # force root to sklearnex
+            kwargs["path"] = [str(pathlib.Path(__file__).parent.parent)]
+        for pkginfo in func(*args, **kwargs):
+            # Do not allow spmd to be yielded
+            if "spmd" not in pkginfo.name.split("."):
+                yield pkginfo
+
+    return wrap
+
+
+def test_all_estimators_covered(monkeypatch):
+    """Check that all estimators defined in sklearnex are available in either the
+    patch map or covered in special testing via SPECIAL_INSTANCES. The estimator
+    must inherit sklearn's BaseEstimator and must not have a leading underscore.
+    The sklearnex.spmd and sklearnex.preview packages are not tested.
+    """
+    monkeypatch.setattr(pkgutil, "walk_packages", _sklearnex_walk(pkgutil.walk_packages))
+    estimators = all_estimators()
+    print(estimators)
+    for name, obj in estimators:
+        # do nothing if defined in preview
+        if "preview" not in obj.__module__:
```

Comment:
Why not use estimator as test parameter and pytest.skip preview ones?

Reply:
Good question, I would have to move the all_estimators monkeypatch into a fixture and then do an indirect parametrization. Objects like BaseSVM could also show up in the list. It will make sure that multiple failures will both show up, not just the first.  If you want me to do it let me know. @Alexsandruss

Reply:
@icfaust, yes, it makes same to do it.

Reply:
Unfortunately using fixtures at collection time is something specifically not supported: https://github.com/pytest-dev/pytest/issues/7140#issuecomment-622237245 In order to do this, I would need to manually monkeypatch instead of using the pytest monkeypatch fixture, which has implications on test isolation. What I will do is change the logic to collect all missing estimators and display them all as a single fail assert.